### PR TITLE
upgrade to yui 3.5.1

### PR DESCRIPTION
**Pull request by drewfish**
Here's an update to Dave's PR #165.  I fixed the failing regression, and ran some sample apps (and fixed issues found during that).

### CODE REVIEW

File: source/lib/app/autoload/loader.common.js [hunk 1]
```diff
@@ -59,23 +59,8 @@ YUI.add('mojito-loader', function(Y, NAME) {
...
-                            done = true;
-                        }
-                    }
-
-                    if (done) {
-                        Y.log('SUCCESS', 'mojito', NAME);
-                        cb(null);
-                    }
+                    Y.log('SUCCESS', 'mojito', NAME);
+                    cb(null);
```
Comment (mojit0) [inline source/lib/app/autoload/loader.common.js]: why the explicit null?

File: source/lib/store.server.js [hunk 2]
```diff
@@ -3733,20 +3731,20 @@ ServerStore.prototype = {
...
 
-        Y = YUI().useSync('loader-base');
-        loader = new Y.Loader({ lang: ctx.lang });
+        Y = YUI({ useSync: true }).use('loader-base');
+        Y.applyConfig({ useSync: false });
 
         // We need to clear YUI's cached dependencies, since there's no
         // guarantee that the previously calculated dependencies have been done
         // using the same context as this calculation.
         delete YUI.Env._renderedMods;
```
Comment (mojit0) [inline source/lib/store.server.js:3741]: missing semi?

Comment (drewfish) [inline source/lib/store.server.js:3741]: I just fixed that.  Refresh the PR.

File: source/lib/app/autoload/loader.common.js (hunk 1 above)
Comment (drewfish) [inline source/lib/app/autoload/loader.common.js]: It's not strictly necessary.  The first argument of the callback is an error object, and by using `cb(null)` I'm say "really really there's no error".  A matter of style I guess.  I can remove it.

Comment (mojit0) [inline source/lib/app/autoload/loader.common.js]: As a matter of style I like to keep semantics between null and undefined clear. Passing null here means arguments.length == 1 on the other side. That's different than allowing it to remain 'undefined'.

**Comment by mridgway** [comment]
Nice.

**Pull request merged**